### DISCUSSION TITLE
Windows Pending Agents Status

### DISCUSSION THREAD

**Discussion by samuelstephens** [wazuh/wazuh#13959] [Q&A] [answered]
Having an issue with my Wazuh cluster where my agents hang in a pending state every 15 minutes.
It used to be they'd disconnect with the default queue size however maxing this out has gotten me to this point where I can see the dip in pending agents every 15 minutes.

Has anybody had a similar experience with this? 90% of the environment is Windows agents.
I have also fiddled with the agent_disconnection_time (10m) and agent_disconnection_alert_time (1h) to try and fix the issue.

SCA hangs and Inventory data seems to timeout at 20000ms so could it possibly be the interval of this check causing the agent changes?

**Comment by mdiego92** [comment]
Hello Samuel Stephens,

Could you please tell us what Wazuh version you are using in your manager and in your agents?

Please refer to the following documentation: [Agent Life Cycle](https://documentation.wazuh.com/current/user-manual/agents/agent-life-cycle.html) 
When the authentication process is pending it means that the manager has received a request for connection from the agent but has not received anything else. This may indicate a firewall issue. The agent will be in this state one time in its life cycle after each startup.  

Keep in mind that the port to register the agent (by default) is 1515 but the communication port (by default) is 1514, so if you have opened only the port 1515 in the firewalls, the agents will be able to register and get a key from the manager but they won't be able to communicate.  

Please, check if port 1514 is opened on the manager side. To do this type:
`telnet MANAGER-IP 1514`

Thank you very much.

**Reply by samuelstephens** [reply]
Hi mdiego92,

The Wazuh manager is operating at v4.3.1, the Windows agents are operating at v4.2.5 with an upgrade to 4.3.1 on the way (please let me know if this version would provide some decent fixes). As for ports I'll have to get back to you as I'm not sure telnet is enabled on the manager. 

I would also note that I'm running it clustered in Kubernetes. Adding more worker nodes seems to have stabilised the connection states a bit however there is still some slight fluctuation surrounding a few agents. Noting the default for syscollector is set to 1h, is it worth setting this to 1d so that it runs less frequently?

Thanks,
Sam

**Comment by mdiego92** [comment, marked as answer]
Hello samuelstephens,

The Wazuh V4.3.1 has a few fixes regarding the previous versions. Please refer to the following article for the release notes: https://documentation.wazuh.com/current/release-notes/release-4-3-1.html

Please check the required ports for the Wazuh services:
![image](https://user-images.githubusercontent.com/62477966/177849135-7b9a583f-f3ed-4483-9262-0bfe2ebf7a7b.png)
https://documentation.wazuh.com/current/getting-started/architecture.html#required-ports

Changing the <interval> setting for `syscollector` could impact positively the performance, but we would like to further analyze the issue before suggesting any changes. Could you please provide us with the following information?

- How many agents are having this issue? How many agents do you have in total?
- How many Wazuh nodes do you have?

Thank you very much.
